src/app.js

```diff
@@ -452,18 +452,32 @@ var run = function () {
 
   var previousInput = ''
   var compileTimeout = null
+  var saveTimeout = null
 
   function editorOnChange () {
     var input = editor.getValue()
-    if (input === '') {
-      editor.setCacheFileContent('')
-      return
-    }
+
+    // if there's no change, don't do anything
     if (input === previousInput) {
       return
     }
     previousInput = input
 
+    // fire storage update
+    // NOTE: save at most once per 5 seconds
+    if (saveTimeout) {
+      window.clearTimeout(saveTimeout)
+    }
+    saveTimeout = window.setTimeout(function () {
```

Comment:
In order to use `setInterval` we need to cancel it every time we switch tabs.  I think we should redesign the editor class and the events to make this more effective.

I would also like to remove the editor dependency in the compiler.